src/Classes/ItemsTab.lua

```diff
@@ -2191,16 +2215,25 @@ function ItemsTabClass:AddCustomModifierToDisplayItem()
 				end
 				return modA.level > modB.level
 			end)
-		elseif sourceId == "VEILED" then
+		elseif currentSourceId == "VEILED" then
 			for i, mod in pairs(self.build.data.veiledMods) do
 				if self.displayItem:GetModSpawnWeight(mod) > 0 then
-					t_insert(modList, {
-						label =  table.concat(mod, "/") .. " (" .. mod.type .. ")",
+					local currMod = {
```

Comment:
You don't need the big if statement and creation of a local table here.  You can just add a ternary to the statement that was previously there, i.e.
```lua
t_insert(modList, {
	label =  table.concat(mod, "/") .. currentModId == "MODIFIERS" and " (" .. mod.type .. ")" or "",
```